automatic testing script and database

NicoCarpe opened the pull request.

NicoCarpe committed:
automatic testing script and database

NicoCarpe commented:
merging testing and master branch

kylemath reviewed (approved):
Looks great @NicoCarpe , now do we visualize the results easily? 
If I understand you ran a single layer neural network with varying units. This is equivalent (since it is only one layer) to a regression with more and more parameters, 

what range of values are you using for the number of units? 

Can you plot the test accuracy and test loss as a function of number of units?

@korymath - any other ideas?

NicoCarpe commented:
I'm using 2^x , x=0,1,...,6 for the units and I am making sure that if the
units have the form like [2^x,2^x-1,2^x] where the outside terms are always
one degree higher.

And I was thinking the next move would be to query the the results such as
accuracy and loss against units. However I need to find a way to store
units in a way that's not a string so we can properly graph them.

Also I have been using more layers as well and think it might be useful to
graph the effects the number of layers have on the accuracy and loss as
well.

On Tue., May 28, 2019, 19:52 Kyle Mathewson, <notifications@github.com>
wrote:

> *@kylemath* approved this pull request.
>
> Looks great @NicoCarpe <https://github.com/NicoCarpe> , now do we
> visualize the results easily?
> If I understand you ran a single layer neural network with varying units.
> This is equivalent (since it is only one layer) to a regression with more
> and more parameters,
>
> what range of values are you using for the number of units?
>
> Can you plot the test accuracy and test loss as a function of number of
> units?
>
> @korymath <https://github.com/korymath> - any other ideas?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kylemath/DeepEEG/pull/30?email_source=notifications&email_token=AKBLWR7MK3A555LOZRWWGODPXXOUTA5CNFSM4HQH332KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBZ57XDY#pullrequestreview-243006351>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKBLWRY23QP4WBA3SI3AZQ3PXXOUTANCNFSM4HQH332A>
> .
>

kylemath commented:
sounds great, yes manipulating layers is a good idea too,

there is no need to have fewer units in the inner layers, so feel free to
varry that as well,


On Tue, May 28, 2019 at 10:00 PM Nicolas Carpenter <notifications@github.com>
wrote:

> I'm using 2^x , x=0,1,...,6 for the units and I am making sure that if the
> units have the form like [2^x,2^x-1,2^x] where the outside terms are always
> one degree higher.
>
> And I was thinking the next move would be to query the the results such as
> accuracy and loss against units. However I need to find a way to store
> units in a way that's not a string so we can properly graph them.
>
> Also I have been using more layers as well and think it might be useful to
> graph the effects the number of layers have on the accuracy and loss as
> well.
>
> On Tue., May 28, 2019, 19:52 Kyle Mathewson, <notifications@github.com>
> wrote:
>
> > *@kylemath* approved this pull request.
> >
> > Looks great @NicoCarpe <https://github.com/NicoCarpe> , now do we
> > visualize the results easily?
> > If I understand you ran a single layer neural network with varying units.
> > This is equivalent (since it is only one layer) to a regression with more
> > and more parameters,
> >
> > what range of values are you using for the number of units?
> >
> > Can you plot the test accuracy and test loss as a function of number of
> > units?
> >
> > @korymath <https://github.com/korymath> - any other ideas?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/kylemath/DeepEEG/pull/30?email_source=notifications&email_token=AKBLWR7MK3A555LOZRWWGODPXXOUTA5CNFSM4HQH332KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBZ57XDY#pullrequestreview-243006351
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AKBLWRY23QP4WBA3SI3AZQ3PXXOUTANCNFSM4HQH332A
> >
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kylemath/DeepEEG/pull/30?email_source=notifications&email_token=AA36GFKHBELEDUAAC6KN5J3PXX5W7A5CNFSM4HQH332KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWOC5LI#issuecomment-496774829>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA36GFMO4OMJBSW2NY64TZ3PXX5W7ANCNFSM4HQH332A>
> .
>

NicoCarpe commented:
Hi Kyle I was was just letting you know I have class at 10 every day until
1:30 so I cannot make the meeting tomorrow.

NicoCarpe committed:
changed TestDB

Made variable assignment more efficient and created method to organize by layer and units in each layer allowing for easy data analysis.

NicoCarpe committed:
Revised version of TestDB and beggining of Analysis file.

NicoCarpe committed:
SQL statements for analysis + (idea) numpy array for ordering units.

kylemath commented:
Hey @NicoCarpe just saw this old PR, how are you doing lately?

NicoCarpe commented:
Hi, I've been doing alright considering the situation (not the biggest fan of the online class format). How about yourself? 

Also, after you introduced me to this project of yours I became quite interested in computer science applications to neuroscience, but I felt a bit under-equipped to help out properly at the time. In the past few semesters I have taken classes that I felt filled in those gaps of knowledge but I still haven't really gotten my foot in the door in terms of research.
